### ISSUE TITLE
ERROR in ImageminWebpWebpackPlugin: FILE wasn't converted!

### ISSUE THREAD

**Issue by WebTravel** [iampava/imagemin-webp-webpack-plugin#11]
Hello, thanks for your plugin. I use it with default config, but I start my dev server I recieve message
`ERROR in spawn /home/node/node_modules/cwebp-bin/vendor/cwebp ENOENT
ERROR in ImageminWebpWebpackPlugin: "img/example.jpg" wasn't converted!`

```
const ImageminWebpWebpackPlugin = require('imagemin-webp-webpack-plugin');

plugins: [
   new ImageminWebpWebpackPlugin(),
]`

"dev": "webpack-dev-server --mode development --config ./assets/webpack --open --hot"
```

**Comment by iampava** [comment]
Hey! Are you using the lastest version of the plugin?

**Comment by WebTravel** [comment]
Yes, I used last version from npm repository. 
I installed it with next command:
```
yarn add imagemin-webp-webpack-plugin
```
I used [this](https://www.npmjs.com/package/imagemin-webp-webpack-plugin) instruction.

And packages whem I install in my project:
```
"webpack": "^4.20.2",
"webpack-dev-server": "^3.1.4"
"imagemin": "^6.1.0",
"imagemin-webp": "^5.0.0",
"imagemin-webp-webpack-plugin": "^3.1.0",
```

**Comment by iampava** [comment]
Hmm...I double-checked the code and it's probably because `imagemin` failed to convert the image. If you could send it to me I could verify this for sure. Or you can debug it yourself, in case it's a sensitive image which cannot be shared :D

**Comment by WebTravel** [comment]
Do I send image for you?
Plugin console for me that it can't convert any image. But if I run command ```yarn build``` with webpack flag `production` package works correctly. And I see webp images in my folder.
So, I have problem only Dev server and development mode...

**Comment by iampava** [comment]
I see. Yes, send me the image and I'll investigate :) Sorry for the late replies, it's been a hectic week...

**Comment by iampava** [comment]
Closing this due to inactivity. If the problem persists feel free to re-open the issue and I'll do my best to fix it ;)

**Comment by kilgarenone** [comment]
I'm seeing this error when I run in **CircleCI Alpine**. Local build and dev work fine. 

```
ERROR in spawn /root/project/node_modules/cwebp-bin/vendor/cwebp ENOENT

ERROR in ImageminWebpWebpackPlugin: "img/crosssell-default.ec73cd9a4cefd898866ef843c643f55b.png" wasn't converted!
```

**Comment by JamesJGoodwin** [comment]
Same issue here:

```
ERROR in ImageminWebpWebpackPlugin: "assets/img/flags/rs.png" wasn't converted!
ERROR in PNG support not compiled. Please install the libpng development package before building.
```

Installing libs via `apt-get install libpng libpng-dev` is not helping.

**Comment by iampava** [comment]
Hmm... I'm not converting the images myself, I use the [imagemin](https://www.npmjs.com/package/imagemin) and [imagemin-webp](https://www.npmjs.com/package/imagemin-webp) packets for that. 

Looking at my dependencies, I see I don't have their latest version here. I made a task regarding that, I will try and get to it as soon as possible: https://github.com/iampava/imagemin-webp-webpack-plugin/issues/17

**Comment by auliafaizahr** [comment]
> Same issue here:
> 
> ```
> ERROR in ImageminWebpWebpackPlugin: "assets/img/flags/rs.png" wasn't converted!
> ERROR in PNG support not compiled. Please install the libpng development package before building.
> 
> Error! Could not process file /tmp/6e7e3fad-c76e-4189-ac16-77c9879eaa8d
> Error! Cannot read input picture file '/tmp/6e7e3fad-c76e-4189-ac16-77c9879eaa8d 
> ```
> 
> Installing libs via `apt-get install libpng libpng-dev` is not helping.

hi, is this issue is solved ? because i faced the same issue, pls let me know

**Comment by JamesJGoodwin** [comment]
@auliafaizahr try deleting `node_modules` and then just `npm i`. It helped me last time.

**Comment by auliafaizahr** [comment]
@JamesJGoodwin thankyouuu, it works on me too :)

**Comment by longfeiling** [comment]
> @JamesJGoodwin thankyouuu, it works on me too :)



> > Same issue here:
> > ```
> > ERROR in ImageminWebpWebpackPlugin: "assets/img/flags/rs.png" wasn't converted!
> > ERROR in PNG support not compiled. Please install the libpng development package before building.
> > 
> > Error! Could not process file /tmp/6e7e3fad-c76e-4189-ac16-77c9879eaa8d
> > Error! Cannot read input picture file '/tmp/6e7e3fad-c76e-4189-ac16-77c9879eaa8d 
> > ```
> > 
> > 
> > Installing libs via `apt-get install libpng libpng-dev` is not helping.
> 
> hi, is this issue is solved ? because i faced the same issue, pls let me know


> @auliafaizahr try deleting `node_modules` and then just `npm i`. It helped me last time.

why I tried  deleting node_modules and then just npm i，but not use?

**Comment by kv1991** [comment]
Hello, I have the same error right now, anyone can help me?

**Comment by ahmedkhan847** [comment]
anyone facing issue after installing the packages  `apt-get install -y libjpeg-dev libpng-dev libtiff-dev libgif-dev libwebp-dev` remove your node_modules and reinstall packages it will work

**Comment by ismatech** [comment]
hi guys|
I also get this issue but not on a local machine. I get it on CI/CD in GitLab during stage install_script.
And you know there is a build which is being deployed on docker container so the folder `node_modules` is always being removed. Maybe I should install some libs? I've seen one warning/error there: something like "didn't find any C compiler".
![cwebp-bin error](https://github.com/iampava/imagemin-webp-webpack-plugin/assets/30582837/f429ec9c-981a-409f-a6f5-2989334a830f)

**Comment by ismatech** [comment]
I found out why I face this problem. It's because in dockerfile we install from node:14-alpine version. 
If I just write from node:14, then with the command 
```docker
RUN apt update -y && \
    apt install -y  --no-install-recommends \
    ffmpeg && \
    rm -rf /var/lib/apt/lists/*
```
It is being installed correctly. Does anyone know how to install cwebp-bin with node14-alpine version?
UPD: devops guy just created separated dockerfile for node:14 in gitlab ci/cd, so we solved our problem. Hope, it will help you too.
